catalog/spatial/kml/spatial-kml-transformer/src/main/java/org/codice/ddf/spatial/kml/converter/KmlToJtsMultiGeometryConverter.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ **/
+package org.codice.ddf.spatial.kml.converter;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.geotools.geometry.jts.JTSFactoryFinder;
+
+import com.vividsolutions.jts.geom.GeometryFactory;
+
+public class KmlToJtsMultiGeometryConverter {
+    private KmlToJtsMultiGeometryConverter() {
+    }
+
+    public static com.vividsolutions.jts.geom.GeometryCollection from(
+            de.micromata.opengis.kml.v_2_2_0.MultiGeometry kmlMultiGeometry) {
+        if (kmlMultiGeometry == null) {
+            return null;
+        }
+
+        GeometryFactory geometryFactory = JTSFactoryFinder.getGeometryFactory();
+
+        List<com.vividsolutions.jts.geom.Geometry> jtsGeometries = kmlMultiGeometry.getGeometry()
+                .stream()
+                .map(KmlToJtsGeometryConverter::from)
+                .collect(Collectors.toList());
+
+        if (CollectionUtils.isNotEmpty(jtsGeometries)) {
+            return geometryFactory.createGeometryCollection(jtsGeometries.toArray(new com.vividsolutions.jts.geom.Geometry[jtsGeometries.size()]));
```

Comment:
I think it's more performant to use `toArray` with a size 0 array

Reply:
Changed